internal/controllers/device_data_controller.go

```diff
@@ -253,6 +262,54 @@ func (udc *UserDevicesController) RefreshUserDeviceStatus(c *fiber.Ctx) error {
 	return fiber.NewError(fiber.StatusBadRequest, "no active Smartcar integration found for this device")
 }
 
+// QueryDeviceErrorCodes godoc
+// @Description Queries chatgpt for user device error codes
+// @Tags        user-devices
+// @Param       user_device_id path string true "user device ID"
+// @Param       queryDeviceErrorCodes body controllers.QueryDeviceErrorCodesReq true "error codes"
+// @Success     200 {object} controllers.QueryDeviceErrorCodesResponse
+// @Security    BearerAuth
+// @Router      /user/devices/{userDeviceID}/error-codes [post]
+func (udc *UserDevicesController) QueryDeviceErrorCodes(c *fiber.Ctx) error {
+	udi := c.Params("userDeviceID")
+	userID := helpers.GetUserID(c)
+
+	ud, err := models.UserDevices(
+		models.UserDeviceWhere.ID.EQ(udi),
+		models.UserDeviceWhere.UserID.EQ(userID),
+	).One(c.Context(), udc.DBS().Reader)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return fiber.NewError(fiber.StatusNotFound, err.Error())
+		}
+		return err
+	}
+
+	dd, err := udc.DeviceDefSvc.GetDeviceDefinitionByID(c.Context(), ud.DeviceDefinitionID)
+	if err != nil {
+		return helpers.GrpcErrorToFiber(err, "deviceDefSvc error getting definition id: "+ud.DeviceDefinitionID)
+	}
+
+	req := &QueryDeviceErrorCodesReq{}
+	if err := c.BodyParser(req); err != nil {
+		return fiber.NewError(fiber.StatusBadRequest, err.Error())
+	}
+	if err != nil {
+		return fiber.NewError(fiber.StatusBadRequest, err.Error())
+	}
+
+	oi := services.NewOpenAI(*udc.Settings)
```

Comment:
My guess is we're doing this because adding another dependency to the monocontroller is painful.

How testable do you figure this is?